src/hub_api.rs

```diff
@@ -87,6 +90,15 @@ impl IndexEntry {
     pub fn title_words(&self) -> Vec<String> {
         self.title.split_whitespace().map(|t| t.to_lowercase()).collect_vec()
     }
+    pub fn url(&self) -> String {
```

Comment:
Nit: It's more usual for functions like this (and the next two) to return `&str` and the body is `&self.url`.  Let the consumer `clone/to_owned` if they need it.